### ISSUE TITLE
[java] ArrayIsStoredDirectly doesn't consider nested classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3613]
**Affects PMD Version:** 6.40.0

**Rule:** [ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:**

When inner/nested classes are used, then these cases are not detected.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public class Inner {
        String [] arr;
        void foo (String[] x) {arr = x;} // expected violation
    }
}
```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

Note - found via #2687
